frontend/src/old-pages/Clusters/__tests__/util.test.tsx

```diff
@@ -74,3 +85,110 @@ describe('Given a function to get script names of all custom actions', () => {
     })
   })
 })
+
+describe('Given a function to select the current cluster and a cluster name', () => {
+  let mockCluster: MockProxy<ClusterInfoSummary>
+  let mockNavigate: jest.Mock
+  let mockGetConfiguration: jest.Mock
+  let mockDescribeCluster: jest.Mock
+
+  beforeEach(() => {
+    mockNavigate = jest.fn()
+    mockGetConfiguration = jest.fn()
+    mockDescribeCluster = jest.fn(async () => ({}))
+
+    mockCluster = {
+      clusterName: 'some-name',
+      region: 'some-region',
+      version: 'some-version',
+      cloudformationStackArn: 'some-arn',
+      clusterStatus: ClusterStatus.CreateFailed,
+      cloudformationStackStatus: CloudFormationStackStatus.CreateComplete,
+    }
+
+    mockGetState.mockReturnValue([mockCluster])
+  })
+
+  const canFetchCases: ClusterStatus[] = [
+    ClusterStatus.CreateInProgress,
+    ClusterStatus.CreateComplete,
+    ClusterStatus.DeleteInProgress,
+    ClusterStatus.DeleteFailed,
+    ClusterStatus.UpdateInProgress,
+    ClusterStatus.UpdateFailed,
+    ClusterStatus.UpdateComplete,
+  ]
+
+  describe.each(canFetchCases)('when the cluster is in status %s', status => {
+    beforeEach(() => {
+      mockCluster.clusterStatus = status
+      selectCluster(
+        mockCluster.clusterName,
+        mockNavigate,
+        mockGetConfiguration,
+        mockDescribeCluster,
+      )
+    })
+
+    it('should get the cluster configuration', () => {
+      expect(mockGetConfiguration).toHaveBeenCalledTimes(1)
+      expect(mockGetConfiguration).toHaveBeenCalledWith(
+        mockCluster.clusterName,
+        expect.any(Function),
+      )
+    })
+
+    it('should describe the cluster', () => {
+      expect(mockDescribeCluster).toHaveBeenCalledTimes(1)
+      expect(mockDescribeCluster).toHaveBeenCalledWith(
+        mockCluster.clusterName,
+        expect.any(Function),
+      )
+    })
+  })
+
+  const cannotFetchCases: ClusterStatus[] = [
+    ClusterStatus.CreateFailed,
+    ClusterStatus.DeleteComplete,
+  ]
+
+  describe.each(cannotFetchCases)(
+    'when the cluster is in status %s',
+    status => {
+      beforeEach(() => {
+        mockCluster.clusterStatus = status
+        selectCluster(
+          mockCluster.clusterName,
+          mockNavigate,
+          mockGetConfiguration,
+          mockDescribeCluster,
+        )
+      })
+
+      it('should not get the cluster configuration', () => {
+        expect(mockGetConfiguration).toHaveBeenCalledTimes(0)
+      })
+
+      it('should not describe the cluster', () => {
+        expect(mockDescribeCluster).toHaveBeenCalledTimes(0)
+      })
+    },
+  )
+
+  describe('when describing the cluster fails', () => {
```

Comment:
question(non-blocking): I don't see a test in which Describe succeeds and GetConfiguration fails, but I guess test surface here is almost zero right?

Reply:
At the moment there is no specific action to take in that scenario (it would be handled as any other request failure, meaning a red alert with its error message)

Would you treat that case in a specific way? What should the app when the cluster can be described, but its configuration fetch failed?

Reply:
I'd consider that a "no test because not enough test surface"